compiler-core/src/ast.rs

```diff
@@ -1159,29 +1149,35 @@ impl Default for Inferred<()> {
     }
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum AssignVariableType {
+    Alias,
+    Original,
+}
+
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub enum AssignName {
-    Variable(SmolStr),
+    Variable(SmolStr, SrcSpan, AssignVariableType),
```

Comment:
Please do not change this type. This new value doesn't make sense in any of the other places this is used and it is again a duplication of information that is already in the import struct. There must be no duplication of any data that could result in invalid states being representable.

Reply:
I was a bit confused, hopefully not anymore!